src/main/java/org/galatea/starter/service/FinnhubService.java

```
@@ -37,14 +41,49 @@ public List<FinnhubSymbol> getAllSymbols() {
     }
 
     public FinnhubQuote getQuote(String symbol) {
+
         FinnhubQuoteEntity cached = repository.findById(symbol).orElse(null);
+        FinnhubQuote apiQuote = null;
 
-        //Return cached entry if not stale
+        //1. Returned cached data if not stale
         if (cached != null && Instant.now().isBefore(cached.getCachedAt().plus(CACHE_TTL))) {
+            log.info("Retrieved quote from cache: {}", cached.getQuote());
             return cached.getQuote();
         }
 
-        FinnhubQuote apiQuote = finnhubClient.getQuote(symbol);
+        try {
+            //2. Try API
+            apiQuote = finnhubClient.getQuote(symbol);
+
+            //Finnhub will return dummy data with all 0s if the symbol is not found, we want to flag that
+            boolean allZero = apiQuote.getCurrent().compareTo(BigDecimal.ZERO) == 0 &&
+                    apiQuote.getPreviousClose().compareTo(BigDecimal.ZERO) == 0 &&
+                    apiQuote.getHigh().compareTo(BigDecimal.ZERO) == 0 &&
+                    apiQuote.getLow().compareTo(BigDecimal.ZERO) == 0 &&
+                    apiQuote.getOpen().compareTo(BigDecimal.ZERO) == 0;
+
+            //3. Ensure data received is legit
+            if(allZero) {
+                log.error("Finnhub returned no data for symbol: " + symbol);
+                throw new ResponseStatusException(
+                        HttpStatus.BAD_REQUEST,
+                        "Finnhub returned no data for symbol: " + symbol
+                );
+            }
+
+            log.info("Retrieved quote from Finnhub: {}", apiQuote);
+        } catch (ResponseStatusException e) {
+            //Pass through known symbol error
+            throw e;
```

This catch block is a no op since we catch all exceptions in the block below and we're not actually doing anything with the caught exception.